tarofreighter/chain_porter.go

```diff
@@ -509,27 +509,49 @@ func createDummyOutput() *wire.TxOut {
 	return &newOutput
 }
 
-// adjustFundedPsbt takes a PSBT which may have used BIP 69 sorting, and
+// adjustFundedPsbt takes a funded PSBT which may have used BIP 69 sorting, and
 // creates a new one with outputs shuffled such that the change output is the
 // last output.
-func adjustFundedPsbt(pkt *psbt.Packet, changeIndex uint32,
-	anchorInputValue int64) {
+func adjustFundedPsbt(pkt *tarogarden.FundedPsbt, anchorInputValue int64) {
+	// If there is no change there's nothing we need to do.
+	changeIndex := pkt.ChangeOutputIndex
+	if changeIndex == -1 {
+		return
+	}
+
 	// Store the script and value of the change output.
-	changeOutput := pkt.UnsignedTx.TxOut[changeIndex]
-	changeAmtSats := changeOutput.Value
-	changeScript := changeOutput.PkScript
+	maxOutputIndex := len(pkt.Pkt.UnsignedTx.TxOut) - 1
+	changeOutput := pkt.Pkt.UnsignedTx.TxOut[changeIndex]
 
 	// Overwrite the existing change output, and restore in at the
 	// highest-index output.
-	maxOutputIndex := len(pkt.UnsignedTx.TxOut) - 1
-	pkt.UnsignedTx.TxOut[changeIndex] = createDummyOutput()
-	pkt.UnsignedTx.TxOut[maxOutputIndex].PkScript = changeScript
-	pkt.UnsignedTx.TxOut[maxOutputIndex].Value = changeAmtSats
+	pkt.Pkt.UnsignedTx.TxOut[changeIndex] = createDummyOutput()
+	pkt.Pkt.UnsignedTx.TxOut[maxOutputIndex].PkScript = changeOutput.PkScript
+	pkt.Pkt.UnsignedTx.TxOut[maxOutputIndex].Value = changeOutput.Value
 
 	// Since we're adding the input of the anchor output of our prior asset
 	// later, we need to add this value here, so we don't lose the amount
 	// to fees.
-	pkt.UnsignedTx.TxOut[maxOutputIndex].Value += anchorInputValue
+	pkt.Pkt.UnsignedTx.TxOut[maxOutputIndex].Value += anchorInputValue
+
+	// If the change output already is the last output, we don't need to
+	// overwrite anything in the PSBT outputs.
+	if changeIndex == int32(maxOutputIndex) {
```

Comment:
This is a bit confusing. Could this check be done earlier in the function OR removed (and have the below operations just be no-ops.

Reply:
I think it could be done earlier, but when `changeIndex == maxOutputIndex` we still need to increment the value of the last output / change output by `anchorInputValue`.

Reply:
FWIW, we'll be heavily refactoring this area as part of #122